tracer/src/Datadog.FleetInstaller/GacInstaller.cs

```diff
@@ -147,43 +148,102 @@ private static unsafe bool TryGacUninstall(ILogger log, IEnumerable<(string Asse
 
             foreach (var (gacAssemblyPath, associatedFilepath) in filesToRemove)
             {
-                var gacName = Path.GetFileNameWithoutExtension(gacAssemblyPath);
-                if (string.IsNullOrEmpty(gacName))
+                var simpleName = Path.GetFileNameWithoutExtension(gacAssemblyPath);
+                if (string.IsNullOrEmpty(simpleName))
                 {
                     log.WriteError($"Error uninstalling '{gacAssemblyPath}' from GAC: could not determine GAC name.");
                 }
 
-                Fusion.UninstallDisposition disposition = 0;
-                fixed (char* associatedFilepathPtr = associatedFilepath)
+                // We need the full version of the assembly we're uninstalling, but we only have the name
+                // (the file might not exist on disk anymore, so we can't try to read it either). We can't use
+                // the simple name, as if there are multiple versions of the Datadog.Trace (for example)
+                // assembly installed, then the fusion API returns
+                // IASSEMBLYCACHE_UNINSTALL_DISPOSITION_REFERENCE_NOT_FOUND if we try to use the simple name,
+                // even if they have different filepath references.
+                List<string>? assemblyNamesToTry = null;
+                retValue = Fusion.CreateAssemblyNameObject(out var gacAssemblyName, simpleName, Fusion.CREATE_ASM_NAME_OBJ_FLAGS.NONE, IntPtr.Zero);
+                if (retValue != HResult.Code.S_OK || gacAssemblyName is null)
                 {
-                    var fusionInstallReference = new Fusion.FUSION_INSTALL_REFERENCE
-                    {
-                        cbSize = (uint)sizeof(Fusion.FUSION_INSTALL_REFERENCE),
-                        dwFlags = Fusion.FusionInstallReferenceFlags.None,
-                        guidScheme = Fusion.FUSION_INSTALL_REFERENCE.FUSION_REFCOUNT_FILEPATH_GUID,
-                        szIdentifier = associatedFilepathPtr
-                    };
-                    retValue = ppAsmCache.UninstallAssembly(0U, gacName, &fusionInstallReference, &disposition);
-                }
-
-                if (disposition is Fusion.UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_ALREADY_UNINSTALLED
-                    or Fusion.UninstallDisposition.IASSEMBLYCACHE_UNINSTALL_DISPOSITION_REFERENCE_NOT_FOUND)
-                {
-                    log.WriteInfo($"Assembly '{gacAssemblyPath}' was already uninstalled from the GAC.");
+                    log.WriteError($"Error creating IAssemblyName object for {simpleName}. Error code {retValue} returned from CreateAssemblyEnum. Continuing with simple assembly name {simpleName}");
```

Comment:
I feel like this should be a warning since we're not giving up yet

Reply:
We don't have warnings currently, and I couldn't be bothered to create them just for this 🙈

Reply:
Fine, your silence was deafening. I've added Warning